[Auto Parallel] Upgrade fluid comm operators to be compatible with new comm library

GhostScreaming opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
Others

### PR changes
Others

### Description
Pcard-70448

Based on compatible new communication library upgrade PR https://github.com/PaddlePaddle/Paddle/pull/56604. Old communication library is used in default, export `FLAGS_dynamic_static_unified_comm=1` to use new communication library.

Following operators are upgraded compatibly  now: `alltoall`, `barrier`, `c_concat`, `c_allreduce`, `c_sync_stream_op`, `c_wait_comm_op`, `c_wait_compute_op`, `recv_v2_op`, `send_v2_op`, `c_reduce_op`, `c_allgather_op`, `c_scatter_op`, `c_reducescatter_op`.

wentaoyu committed:
for verify

fluid operator support new comm library

wentaoyu committed:
u

wentaoyu committed:
u

wentaoyu committed:
u

GhostScreaming committed:
compatiable new comm library upgrade for c_allgather, c_reduce, c_reduce_scatter and c_scatter.

hitywt commented on python/paddle/distributed/auto_parallel/static/process_group.py:
```diff
@@ -194,28 +213,29 @@ def instantiate(self):
-            paddle._legacy_C_ops.barrier(
-                barrier_tensor, barrier_tensor, 'ring_id', 0
-            )
+     #      barrier_tensor = paddle.full([1], 1, dtype="int32")
```
这里注释可以取消掉测试下，注释掉可能会跳过一些问题

GhostScreaming committed:
Remove useless comments in process_group.py

GhostScreaming committed:
Polish code style.

GhostScreaming committed:
Fix some problems.

GhostScreaming committed:
Remove use fluid api in phi comm_context_manager.

GhostScreaming committed:
Add PPADDLE_WITH_CUDA and PADDLE_WITH_NCCL micro judgement.

hitywt commented on paddle/phi/core/distributed/store/tcp_store.cc:
```diff
@@ -351,12 +351,20 @@ TCPStore::TCPStore(std::string host,
+            << ", timeout: " << timeout;
   }
 
+  VLOG(0) << "bfore connect host " << host << ", port:" << port;
```
VLOG(0)都删除下吧

hitywt commented on paddle/fluid/operators/collective/barrier_op.cu.cc:
```diff
@@ -38,13 +40,45 @@ class BarrierOpCUDAKernel : public framework::OpKernel<T> {
+        phi::distributed::CommContextManager::GetInstance();
+    VLOG(0) << "new comm_context_manager has rid " << rid;
+    if (comm_context_manager.Has(std::to_string(rid))) {
+      VLOG(0) << "new comm_context_manager has rid " << rid;
```
这里VLOG(0)也是调试用的，可以删除

hitywt commented on paddle/phi/core/distributed/comm_context_manager.cc:
```diff
@@ -64,7 +73,61 @@ void CommContextManager::CreateNCCLCommContext(
+    // *******************************************************************************
+    // dev_ctx->SetAllocator(
+    //     paddle::memory::allocation::AllocatorFacade::Instance()
+    //         .GetAllocator(phi::GPUPlace(CommContextManager::device_id),
```
注释代码可以删除了

hitywt commented on paddle/phi/core/distributed/nccl_comm_context.cc:
```diff
@@ -37,6 +37,36 @@ NCCLCommContext::NCCLCommContext(int rank, int size, ncclUniqueId nccl_id)
 
+gpuStream_t NCCLCommContext::GetStream() {
+  if (!dev_ctx_) {
+    VLOG(0) << "NCCLCommContext::GetStream dev_ctx nullptr";
```
调试VLOG(0)需要删除

hitywt commented on python/paddle/distributed/auto_parallel/static/process_group.py:
```diff
@@ -58,6 +59,7 @@ def remove_process_group(ring_id):
 def new_process_group(
     ranks, group_id=None, force_new_group=False, group_type=None
 ):
+    logger.info(f"debug new_process_group group_id: {group_id}")
```
调试log，可以删除

GhostScreaming committed:
Fix bug of HIP architecture.

GhostScreaming committed:
Fix some problems.
1. remove useless loggings.
2. Fix conditional compilation for HIP.
3. Fix problems of test_pass_generation_pipeline.py. It calls paddle.distributed.init_parallel_env() at first,
then auto.Engine calls _init_comm(), which will calls process_group.instantiate(). However, init_parallel_env() will call
paddle.distributed.barrier(), it will call CreateNCCLEnvCache and create corresponding NCCLCommContext. But dev_id is not
set, as a result, NCCLCommContext's dev_ctx is not initialized.

GhostScreaming committed:
Fix some problems.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into static_compat_upgrade

GhostScreaming committed:
Polish code.

GhostScreaming committed:
Polish code.

GhostScreaming committed:
Revert compatiable upgrade for communication operators. Their upgrades
will be submitted in another PR.

GhostScreaming committed:
Remove StaticTCPStore.

GhostScreaming committed:
Remove useless modification.

GhostScreaming committed:
Remove useless set_cuda_device_id.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into pure_static_compat_upgrade

GhostScreaming committed:
Polish code.

GhostScreaming committed:
Remove fluid header files in phi files.

GhostScreaming committed:
Remove useless comments.

GhostScreaming committed:
Fix problems of hip arch.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into pure_static_compat_upgrade

GhostScreaming committed:
Fix some problems.

GhostScreaming committed:
Polish code.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into pure_static_compat_upgrade

GhostScreaming committed:
Polish code style.

GhostScreaming committed:
Verify compatiable comm lib upgrade in 1.3B GPT-3 DP8 training.

GhostScreaming committed:
Merge branch 'pure_static_compat_upgrade' of https://github.com/GhostScreaming/Paddle into static_compat_upgrade

GhostScreaming committed:
Verify fluid communication upgrade in dp8, mp8 and pp8.

GhostScreaming committed:
Use Paddle Flags to control fluid communication operators.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into static_compat_upgrade

GhostScreaming committed:
Add testcases and fix some problems in fluid comm ops.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into static_compat_upgrade

GhostScreaming committed:
Change usage of Paddle flags.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into static_compat_upgrade

GhostScreaming committed:
Remove useless codes.

hitywt commented on test/collective/test_collective_sendrecv.py:
```diff
@@ -25,18 +25,36 @@ class TestSendRecvOp(TestDistBase):
+    #     self.check_with_place(
+    #         "collective_sendrecv_op_array.py", "sendrecv_array"
+    #     )
+
```
这里为什么删除了

GhostScreaming replied:
之前的注释忘记删了，已经修复

hitywt reviewed (approved):
LGTM

GhostScreaming committed:
Fix some bugs of testcases.

zhiqiu reviewed (dismissed):
LGTM

GhostScreaming committed:
Polish testcases.

GhostScreaming committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into static_compat_upgrade

GhostScreaming committed:
Remove fluid APIs.

luotao1 reviewed (approved):
LGTM for flags

XieYunshen reviewed (approved):
LGTM

zhiqiu reviewed (approved):
LGTM
